packages/zoe/src/contracts/barterExchange.js

```diff
@@ -1,136 +1,141 @@
 // @ts-check
 
 import harden from '@agoric/harden';
+import makeStore from '@agoric/store';
 import { makeZoeHelpers, defaultAcceptanceMsg } from '../contractSupport';
 
-/** @typedef {import('../zoe').ContractFacet} ContractFacet */
-
 /**
+ * This barter exchange accepts offers to trade arbitrary goods for other
+ * things. It doesn't require registration of Issuers. If two offers satisfy
+ * each other, the amounts specified in each side's `want` clause are exchanged.
+ *
  * The Barter Exchange only accepts offers that look like
  * { give: { In: amount }, want: { Out: amount} }
- *
  * The want and give amounts are both treated as minimums. Each successful
  * trader gets their `want` and may trade with counter-parties who specify any
  * amount up to their specified `give`.
+ *
+ * @typedef {import('../zoe').ContractFacet} ContractFacet
+ * @param {ContractFacet} zcf
  */
-export const makeContract = harden(
-  /** @param {ContractFacet} zcf */ zcf => {
-    // bookOrders is a Map of Maps. The first key is the brand of the offer's
-    // GIVE, and the second key is the brand of their WANT. For each offer, we
-    // store the handle and the amounts for `give` and `want`.
-    const bookOrders = new Map();
-
-    const { canTradeWithMapKeywords } = makeZoeHelpers(zcf);
-
-    function lookupBookOrders(brandIn, brandOut) {
-      let ordersMap = bookOrders.get(brandIn);
-      if (!ordersMap) {
-        ordersMap = new Map();
-        bookOrders.set(brandIn, ordersMap);
-      }
-      let ordersArray = ordersMap.get(brandOut);
-      if (!ordersArray) {
-        ordersArray = [];
-        ordersMap.set(brandOut, ordersArray);
-      }
-      return ordersArray;
-    }
-
-    function findMatchingTrade(newDetails, orders) {
-      return orders.find(order => {
-        return canTradeWithMapKeywords(
-          newDetails.offerHandle,
-          order.offerHandle,
-          [
-            ['In', 'Out'],
-            ['Out', 'In'],
-          ],
-        );
-      });
-    }
+const makeContract = zcf => {
+  // bookOrders is a Map of Maps. The first key is the brand of the offer's
+  // GIVE, and the second key is the brand of their WANT. For each offer, we
```

Comment:
"its WANT" instead of "their WANT"? Since you've established "the offer" is singular.  Also, it looks like you may be distinguishing give and want in different ways in different comments; "give" vs. GIVE. If so, might want to standarize.